### PR TITLE
458 table scalability

### LINKED ISSUE #458: [Table] scalability to millions of rows and up to 500 columns
[Internal Watson-IoT/pal-tracking#409](https://github.ibm.com/Watson-IoT/pal-tracking/issues/409)

We need clarity on the specifics of this ask. From a UX standpoint, usability would dictate that we would never surface this many rows or columns to the user. 

Clarifications of the ask should be posted in the thread and any eventual ux design work should be added here to inform the development of this feature.

### PR DESCRIPTION
**Pull request by davidicus**
Closes #458 

**Summary**

- There is an issue with our pagination that when the table is loaded with a large data set it will try to make select options for the dropdown for each page. This causes the table to crash and never render. 

**Change List (commits, features, bugs, etc)**

- Introduction of a `maxPages` prop that is passed along with the `view.pagination` object. This prop will limit the amount of pages that the dropdown will try to render. 
- Updated story to include a maxPages prop so we can see the page limit in the story. 

**Acceptance Test (how to verify the PR)**

- Go to [Table story](?path=/story/watson-iot-table--stateful-example-with-expansion-and-column-resize)
- see the pagination size is limited to 2 because of filters.
- Remove filters and see that maxPages is limiting the amount of pages to 3.

### CODE REVIEW

File: src/components/Table/Table.jsx [hunk 1]
```diff
@@ -568,7 +572,11 @@ const Table = props => {
       visibleData &&
       visibleData.length ? ( // don't show pagination row while loading
         <Pagination
-          {...view.pagination}
+          pageSize={paginationProps.pageSize}
+          pageSizes={paginationProps.pageSizes}
+          page={paginationProps.page}
+          isItemPerPageHidden={paginationProps.isItemPerPageHidden}
+          totalItems={paginationProps.totalItems < maxPages ? paginationProps.totalItems : maxPages}
```
Comment (tay1orjones) [inline src/components/Table/Table.jsx:579]: Could we cover this in a test? 1001 pages worth of data is given to the table, then assert the select contains only 1000 items.

Comment (davidicus) [reply src/components/Table/Table.jsx:579]: Done

**Comment by davidicus** [comment]
Test added.

**Comment by davidicus** [comment]
@stuckless

File: src/components/Table/Table.story.jsx [hunk 2]
```diff
@@ -511,11 +511,18 @@ storiesOf('Watson IoT/Table', module)
...
+    'Stateful Example with expansion, maxPages, and column resize',
     () => (
       <FullWidthWrapper>
         <StatefulTable
           {...initialState}
+          view={{
+            ...initialState.view,
+            pagination: {
+              ...initialState.view.pagination,
+              maxPages: 30,
```
Comment (bjornalm) [inline src/components/Table/Table.story.jsx:523]: We could add a story knob for this

File: src/components/Table/Table.jsx (hunk 1 above)
Comment (stuckless) [inline src/components/Table/Table.jsx:579]: Is the calculation correct?   Should it be
```
totalItems={paginationProps.totalItems < (maxPages * paginationProps.pageSize) ? paginationProps.totalItems : (maxPages * paginationProps.pageSize)}
```
Is `totalItems` the total number of items or the total number of pages?

Comment (tay1orjones) [reply src/components/Table/Table.jsx:579]: Yeah it looks like `totalItems` should be multiplied against the `pageSize`.

**Pull request merged**